File: php/206-reverse-linked-list.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution 
{
    /**
     * @param ListNode $head
     * @return ListNode
     */
    public function reverseList($head) 
    {
        $cur = $head;
        $prev = null;
        while ($cur->next != null) {
            $next = $cur->next;
            $cur->next = $prev;
            $prev = $cur;
            $cur = $next;
        }
        $cur->next = $prev;
        return $cur;
    }
}

File: go/206-reverse-linked-list.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	var prev, current, next *ListNode
	current = head
	for current.Next != nil {
		next = current.Next
		current.Next = prev
		prev = current
		current = next
	}
	current.Next = prev
	return current
}
